### PR TITLE
Update m3cluster and m3metrics

### PR DESCRIPTION
**Pull request by cw9**
Use kv configuration from m3cluster
Decouple the election kv namespace with the kv config, so they don't have to be the same namespace(although they still can)

@xichen2020 @jeromefroe

### CODE REVIEW

File: services/m3aggregator/config/aggregator.go [hunk 1]
```diff
@@ -120,8 +120,11 @@ type AggregatorConfiguration struct {
 	// Client configuration for key value store.
 	KVClient kvClientConfiguration `yaml:"kvClient" validate:"nonzero"`
 
-	// KV namespace.
-	KVNamespace string `yaml:"kvNamespace" validate:"nonzero"`
+	// Election KV namespace.
+	ElectionKVNamespace string `yaml:"electionKVNamespace" validate:"nonzero"`
```
Comment (xichen2020) [inline services/m3aggregator/config/aggregator.go]: Remove this and simply use KVConfig.Namespace below?

Comment (cw9) [reply services/m3aggregator/config/aggregator.go]: Feels a little bit weird that way, although right now the namespace is the same for the placement and the election, but I feel they probably don't need to be tied together, also it's a bit weird that only namespace is used but not the environment. I could just use KVConfig.Namespace if you feel strongly.

Comment (xichen2020) [reply services/m3aggregator/config/aggregator.go]: Agreed that the namespaces don't have to be the same, however I'd prefer to have a single namespace in the config until there is a need to have two separate namespaces for placement and election.

Comment (cw9) [reply services/m3aggregator/config/aggregator.go]: Sounds good

**Comment by cw9** [comment]
Sure, will do

**Pull request merged**